src/components/dialog/dialog.spec.js

```diff
@@ -168,6 +168,27 @@ describe('$mdDialog', function() {
       expect(content.id).toBe('dialogContent_' + dialog[0].id);
     }));
 
+    it('should not clobber the id from `md-dialog` when there is no content', inject(function ($mdDialog, $rootScope, $document) {
```

Comment:
The bug is only when the dialog doesn't have an `md-dialog-content`. The fact that the `md-dialog` has an ID is implied by "clobbers" (i.e., overwrites).